game-core/src/main/java/games/strategy/engine/data/changefactory/serializers/PrimitiveAttachable.java

```diff
@@ -0,0 +1,85 @@
+package games.strategy.engine.data.changefactory.serializers;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.GamePlayer;
+import games.strategy.engine.data.Named;
+import games.strategy.engine.data.NamedAttachable;
+import games.strategy.engine.data.RelationshipType;
+import games.strategy.engine.data.Resource;
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.TerritoryEffect;
+import games.strategy.engine.data.UnitType;
+import games.strategy.triplea.delegate.TechAdvance;
+import java.io.Serializable;
+
+/**
+ * Serializes NamedAttachments into a primitive form that can be deserialized with access to
+ * GameData
+ */
+public class PrimitiveAttachable implements Serializable {
```

Comment:
PrimitiveAttachable is just a primitive version of NamedAttachments?

Reply:
Basically, though I mistyped the class name.  It isn't `NamedAttachment`s but `NamedAttachable`s.

It appears that all NamedAttachable are constant and based off of the original game xml.  That is how `GameObjectStreamData` treats them.  When they are serialized by`GameObjectOutputStream` will convert these objects into a `GameObjectStreamData`.  Then, when they are unserialized by `GameObjectInputStream`, it will use the name to look up the object in the local `GameData`.